src/main/java/com/mojang/serialization/MapCodec.java

```diff
@@ -277,11 +277,31 @@ public String toString() {
         });
     }
 
-    static <A> MapCodec<A> unit(final A defaultValue) {
+    public MapCodec<A> setPartial(final Supplier<A> value) {
+        return mapResult(new ResultFunction<A>() {
+            @Override
+            public <T> DataResult<A> apply(final DynamicOps<T> ops, final MapLike<T> input, final DataResult<A> a) {
+                return a.setPartial(value);
+            }
+
+            @Override
+            public <T> RecordBuilder<T> coApply(final DynamicOps<T> ops, final A input, final RecordBuilder<T> t) {
+                return t;
+            }
+
+            @Override
+            public String toString() {
+                // FIXME: toString needs to be lazy everywhere, otherwise suppliers get resolved too early
+                return "SetPartial[" + value + "]";
```

Comment:
Hmm, don't think a supplier's `toString` calls `get` by default? this fixme probably should be moved or removed as this place looks good

Reply:
not the supplier, but codecs might; need to still go over them and check everything, but this is enough for now